libelement.CLI/include/evaluate_command.hpp

```diff
@@ -74,30 +74,33 @@ namespace libelement::cli
             element_object* result_object;
             if (!custom_arguments.arguments.empty())
             {
-                element_object* call_objects;
-                int call_objects_count;
-                result = context->call_expression_to_objects(nullptr, custom_arguments.arguments.c_str(), &call_objects, &call_objects_count);
+                std::vector<element::object_const_shared_ptr> objs;
+                context->call_expression_to_objects(nullptr, custom_arguments.arguments.c_str(), objs);
+
+                element_object** call_objects = new element_object*[objs.size()];
+                const int call_objects_count = static_cast<int>(objs.size());
+
+                for (int i = 0; i < call_objects_count; ++i)
+                    call_objects[i] = new element_object{ std::move(objs[i]) };
+
                 if (result != ELEMENT_OK)
                 {
                     for (int i = 0; i < call_objects_count; ++i)
-                    {
-                        element_object* obj = &call_objects[i];
-                        element_delete_object(&obj);
-                    }
+                        element_delete_object(&call_objects[i]);
+
                     context->global_scope->remove_declaration(element::identifier{ "<REMOVE>" });
                     return compiler_message(error_conversion(result),
                                             "Failed to convert call expression to objects: " + expression + " called with " + custom_arguments.arguments + " at compile-time with element_result " + std::to_string(result),
                                             compilation_input.get_log_json());
                 }
 
                 result = element_object_call(expression_object, compilation_context, call_objects, call_objects_count, &result_object);
+
+                for (int i = 0; i < call_objects_count; ++i)
+                    element_delete_object(&call_objects[i]);
+
```

Comment:
Where are calling delete[] on call_objects?

Reply:
Also, why are you newing stuff up before testing result != ELEMENT_OK? Could just new up before element_object_call

Reply:
pushed a commit to fix the memory leak 😅

Reply:
also changed it so we new after :)